1-js/04-object-basics/06-constructor-new/article.md

```diff
@@ -60,74 +60,86 @@ let user = {
 };
 ```
 
-现在，如果我们想创建其他用户，我们可以调用 `new User("Ann")`，`new User("Alice")` 等等。比每次使用字面量创建要短得多，而且易于阅读。
+现在，如果我们想创建其他用户，我们可以调用 `new User("Ann")`，`new User("Alice")` 等。比每次都使用字面量创建要短得多，而且更易于阅读。
 
-这是构造函数的主要目的 — 实现可重用的对象创建代码
+这是构造函数的主要目的 — 实现可重用的对象创建代码。
 
-让我们再次注意 — 从技术上讲，任何函数都可以用作构造函数。即：任何函数都可以运行 `new`，它会执行上面的算法。 “首字母大写”是一个共同的约定，以明确表示一个函数将被使用 `new` 运行。
+让我们再强调一遍 — 从技术上讲，任何函数都可以用作构造函数。即：任何函数都可以通过 `new` 来运行，它会执行上面的算法。 “首字母大写”是一个共同的约定，以明确表示一个函数将被使用 `new` 来运行。
 
 ````smart header="new function() { ... }"
-如果我们有许多关于创建单个复杂对象的代码行，我们可以将它们封装在构造函数中，如下所示：
+如果我们有许多行用于创建单个复杂对象的代码，我们可以将它们封装在构造函数中，像这样：
 
 ```js
 let user = new function() {
   this.name = "John";
   this.isAdmin = false;
 
-  // ...用户创建的其他代码
-  // 也许是复杂的逻辑和陈述
+  // ……用于用户创建的其他代码
+  // 也许是复杂的逻辑和语句
   // 局部变量等
 };
 ```
 
-构造函数不能被再次调用，因为它不保存在任何地方，只是被创建和调用。所以这个技巧的目的是封装构建单个对象的代码，而不是将来重用。
+构造函数不能被再次调用，因为它不保存在任何地方，只是被创建和调用。因此，这个技巧旨在封装构建单个对象的代码，而无需将来重用。
 ````
 
 ## 双语法构造函数：new.target
 
-在一个函数内部，我们可以使用 `new.target` 属性来检查它被调用时，是否使用了 `new`。
+```smart header="进阶内容"
+本节涉及的语法内容很少使用，除非你想了解所有内容，否则你可以直接跳过该语法。
+```
+
+在一个函数内部，我们可以使用 `new.target` 属性来检查它是否被使用 `new` 进行调用了。
 
-常规调用为空，如果通过 `new` 调用，则等于函数：
+对于常规调用，它为空，对于使用 `new` 的调用，则等于该函数：
 
 ```js run
 function User() {
   alert(new.target);
 }
 
-// 不带 new：
+// 不带 "new"：
+*!*
 User(); // undefined
+*/!*
 
-// 带 new：
+// 带 "new"：
+*!*
 new User(); // function User { ... }
+*/!*
 ```
 
-这可以使 `new` 和常规语法的工作原理相同：
+它可以被用在函数内部，来判断该函数是被通过 `new` 调用的“构造函数模式”，还是没被通过 `new` 调用的“常规模式”。
+
+我们也可以让 `new` 调用和常规调用做相同的工作，像这样：
 
 ```js run
 function User(name) {
-  if (!new.target) { // 如果你没有运行 new
-    return new User(name); // ...会为你添加 new
+  if (!new.target) { // 如果你没有通过 new 运行我
+    return new User(name); // ……我会给你添加 new
   }
 
   this.name = name;
 }
 
-let john = User("John"); // 重新调用 new User
+let john = User("John"); // 将调用重定向到新用户
 alert(john.name); // John
 ```
 
-这种方法有时用在库中以使语法更加灵活。但因为省略 `new` 使得它不易阅读，这可不是一件好事。 而通过 `new` 我们可以都知道这个新对象正在创建。
+这种方法有时被用在库中以使语法更加灵活。这样人们在调用函数时，无论是否使用了 `new`，程序都能工作。
+
+不过，到处都使用它并不是一件好事，因为省略了 `new` 使得很难观察到代码中正在发生什么。而通过 `new` 我们都可以知道这创建了一个新对象。
 
-## 构造函数 Return
+## 构造函数的 Return
 
-通常，构造函数没有 `return` 语句。他们的任务是将所有必要的东西写入 `this`，并自动转换。
+通常，构造函数没有 `return` 语句。它们的任务是将所有必要的东西写入 `this`，并自动转换为结果。
 
-但是，如果有 `return` 语句，那么规则很简单：
+但是，如果这有一个 `return` 语句，那么规则就简单了：
 
-- 如果 `return` 对象，则返回它，而不是 `this`。
-- 如果 `return` 一个原函数，则忽略。
+- 如果 `return` 被一个对象调用，则返回这个对象，而不是 `this`。
+- 如果 `return` 被一个原函数调用，则忽略。
```

Comment:
```suggestion
- 如果 `return` 返回的是一个原始类型，则忽略。
```

Reply:
原文：
- If `return` is called with an object, then the object is returned instead of `this`.
- If `return` is called with a primitive, it's ignored.

我的理解是这样的， if 'return' is called (inside the constructor) to return a primitive type, ：-）